src/Servers/Kestrel/Core/src/Internal/KestrelServerImpl.cs

```diff
@@ -207,7 +208,12 @@ async Task OnBind(ListenOptions options)
             }
             catch (Exception ex)
             {
-                Trace.LogCritical(0, ex, "Unable to start Kestrel.");
+                // Do not log stack trace for known errors https://github.com/dotnet/aspnetcore/issues/29801
+                if (ex is not IOException && ex.InnerException is not AddressInUseException)
+                {
+                    Trace.LogCritical(0, ex, "Unable to start Kestrel.");
+                }
```

Comment:
@davidfowl Was your suggestion to no longer log any startup exceptions? It feels wrong to log everything except AddressInUseExceptions.

Reply:
Yep, don't log anything here. Just rethrow the exception.